codelab-master/apps/playground/src/app/quiz/components/question/question.component.scss

```diff
@@ -0,0 +1,169 @@
+@import "variables";
+
+.options {
+  margin: 0 0 10px 0.4rem;
+  padding: 5px;
+  width: 39.5rem !important;
+
+  mat-radio-button, mat-checkbox {
+    cursor: pointer;
+    width: 39.5rem !important;
+    height: auto !important;
+    font-family: $font-stack;
+    font-size: 20px !important;
+    color: $dark-brown;
+    background-color: $light-gray;
+    border: 2px solid $medium-gray;
+    padding: 10px 5px 5px 30px;
+    margin-left: -5px;
+    vertical-align: middle;
+
+    &:hover {
+      border: 2px solid $vivid-blue;
+    }
+  }
+
+  mat-radio-label, mat-checkbox-label {
+    white-space: normal;
+    margin-top: 5px;
+
+    .feedback-icon {
+      position: absolute;
+      right: 0;
+      margin: 5px 40px 0 0 !important;
+    }
+  }
+
+  .is-correct {
+    background-color: $green;
+  }
+
+  .is-incorrect {
+    background-color: $red;
+  }
+}
+
+.messages {
+  display: flex;
+  justify-content: center;
+  margin-bottom: 10px;
+  height: auto;
+
+  span {
+    font-size: 17px;
+    margin-top: 10px;
+  }
+
+  .sentiment-icon {
+    font-size: 30px;
+    color: $light-yellow;
+    margin: 30px -50px 0 25px;
+    vertical-align: top;
+    float: left;
+  }
+
+  .message {
+    font-family: $font-stack;
+    margin: 20px auto;
+    margin-left: -10px;
+    padding: 10px;
+    width: 36rem;
+    display: inline-flex;
+    justify-content: center;
+    vertical-align: middle;
+  }
+
+  .correct-message {
+    border: 2px solid $vivid-blue;
+    color: $dark-green;
+  }
+
+  .wrong-message {
+    border: 2px solid $red;
+    color: $red;
+    margin-top: -10px;
+
+    .sentiment-icon {
+      float: left;
+      margin-top: -40px;
+    }
+  }
+}
+
+.previous-answer-text {
+  clear: both;
+
+  div {
+    font-family: $font-stack;
+    font-size: 16px;
+    display: flex;
+    justify-content: center;
+
+    span {
+      font-weight: $font-weight-max;
+      font-style: italic;
+    }
+  }
+}
+
+.single-answer {
+  margin-top: 15px;
+
+  mat-radio-button {
+    height: 57.6px;
+  }
+  mat-radio-button:first-child {
+    margin-top: -25px;
+  }
+
+  .message {
+    margin-top: 20px;
+  }
+
+  .sentiment-icon {
+    display: inline;
+    margin-top: 30px;
+  }
+}
+
+.multiple-answer {
+  mat-checkbox {
+    padding-top: 15px;
+    padding-bottom: 15px;
+
+    label .feedback-icon {
+      margin-top: -50px;
+    }
+  }
+  mat-checkbox:first-child {
+    margin-top: -20px !important;
+  }
+
+  .message {
+    margin-top: 30px;
+    margin-bottom: 20px;
+  }
+
+  .feedback-icon {
+    margin-top: 5px !important;
+  }
+
+  .sentiment-icon {
+    margin-top: 40px;
+    margin-left: 15px;
+  }
+}
+
+// hide mat-radio-buttons/mat-checkboxes next to answer choices
+::ng-deep .mat-radio-button .mat-radio-container {
+  display: none;
```

Comment:
Why are we doing it? Can we just ng-if in the template?

Reply:
I tried using <mat-checkbox *ngIf="hasOptions" ...> but doesn't seem to work, not sure how to do this w/o manipulating the DOM...

Reply:
I don't know how to hide the checkbox since the label is nested inside the mat-checkbox. I could use ViewEncapsulation.ShadowDom...

Reply:
I could try using ViewEncapsulation.ShadowDom and some CSS to hide the checkboxes...